slice-core-api/src/main/java/com/cognifide/slice/api/tag/SliceLookupTag.java

```diff
@@ -49,14 +49,17 @@ public void doTag() throws JspException {
 			}
 
 			final PageContext pageContext = (PageContext) getJspContext();
-			final Object model = SliceTagUtils.getFromCurrentPath(pageContext, type, appName);
+			final Class typeClass = SliceTagUtils.getClassFromType(pageContext, type);
+			final Object model = SliceTagUtils.getFromCurrentPath(pageContext, typeClass, appName);
 			pageContext.setAttribute(var, model, PageContext.PAGE_SCOPE);
+		} catch (ClassNotFoundException e) {
+			throw new JspTagException("Could not get class for " + type);
 		} finally {
 			clean();
 		}
 	}
 
-	public void setType(Class<?> type) {
+	public void setType(String type) {

```

Comment:
I've did some searching and trials and errors with code as well, so far what I've got is overloading is not possible for taglibs, also mentioned on [this blog](http://tech.thecoolblogs.com/2010/09/comibmwsjspjspcoreexception-unable-to.html).

I've also did a POC on having a separate 'name' attribute. It works for sure, but is not without it's own repercussions. What if both type and name attribute are used(and say for an edge case, both have different class values), which one should be preferred is the mystery?

Leaving it up to you, should i keep looking for a way to overload the taglib attribute setter or go with a new attribute.